client/src/app/pages/applications/components/application-detail-drawer/review-fields.tsx

```diff
@@ -69,106 +69,86 @@ export const ReviewFields: React.FC<{ application: Application | null }> = ({
-                  {item.type === "App"
-                    ? `App - ${item.name}`
-                    : item.type === "Archetype"
-                    ? `Archetype - ${item.name}`
-                    : "Unknown"}
-                  -{"  "} {item?.review?.businessCriticality || notYetReviewed}
-                </Label>
-              ))
-            : notYetReviewed}
+          {groupedReviewList.length === 0
+            ? notYetReviewed
+            : groupedReviewList.map((item, index) => {
+                const labelText = item?.review?.businessCriticality;
+                return generateReviewLabel(item, labelText);
+              })}
         </DescriptionListDescription>
       </DescriptionListGroup>
       <DescriptionListGroup>
         <DescriptionListTerm>{t("terms.workPriority")}</DescriptionListTerm>
         <DescriptionListDescription cy-data="work-priority">
-          {groupedReviewList.length > 0
-            ? groupedReviewList.map((item, index) => (
-                <Label key={index} className={spacing.mbSm}>
-                  {item.type === "App"
-                    ? `App - ${item.name}`
-                    : item.type === "Archetype"
-                    ? `Archetype - ${item.name}`
-                    : "Unknown"}
-                  -{"  "}
-                  {item?.review?.workPriority || notYetReviewed}
-                </Label>
-              ))
-            : notYetReviewed}
+          {groupedReviewList.length === 0
+            ? notYetReviewed
+            : groupedReviewList.map((item, index) => {
+                const labelText = item?.review?.workPriority;
+                return generateReviewLabel(item, labelText);
+              })}
         </DescriptionListDescription>
       </DescriptionListGroup>
       <DescriptionListGroup>
         <DescriptionListTerm>{t("terms.comments")}</DescriptionListTerm>
         <DescriptionListDescription cy-data="comments">
-          {groupedReviewList.length > 0
-            ? groupedReviewList.map((item, index) => (
-                <>
-                  {item?.review?.comments ? (
-                    <Label key={index} className={spacing.mbSm}>
-                      {item.type === "App"
-                        ? `App - ${item.name}`
-                        : item.type === "Archetype"
-                        ? `Archetype - ${item.name}`
-                        : "Unknown"}
-                      -{"  "}
-                      {item?.review?.comments || " - "}
-                    </Label>
-                  ) : (
-                    "-"
-                  )}
-                </>
-              ))
-            : notYetReviewed}
+          {groupedReviewList.length === 0
+            ? notYetReviewed
+            : groupedReviewList.map((item, index) => {
+                const labelText = item?.review?.comments;
+                return labelText && generateReviewLabel(item, labelText);
+              })}
```

Comment:
If replace the `generateReviewLabel` function with `ReviewLabel` component:

```suggestion
          {groupedReviewList.length === 0
            ? notYetReviewed
            : groupedReviewList.map((item, index) => (
              <ReviewLabel key={index} item={item} labelText={item?.review?.comments} />
            )}
```